### PR TITLE
[libc++] Remove libc++'s own <limits.h>

### PR DESCRIPTION
**Pull request by ldionne**
We do not define anything beyond the C Library's limits.h, so it shouldn't be needed. This removes the need for a GCC-specific workaround, see [1].

[1]: https://gcc.gnu.org/bugzilla/show_bug.cgi?id=107795

Differential Revision: https://reviews.llvm.org/D138384

### CODE REVIEW

**Review by EricWF — APPROVED** [review]

**Comment by EricWF** [comment]
Assuming you get the bots passing, this change seems OK to me.

File: libcxx/include/module.modulemap.in [hunk 1]
```diff
@@ -448,10 +448,6 @@ module std_inttypes_h [system] {
   export *
 }
 // <iso646.h> provided by compiler.
-module std_limits_h [system] {
```
Comment (lygstate) [inline libcxx/include/module.modulemap.in:451]: Whey this removed, because `stdint.h` are also provided by compiler

Comment (ldionne) [reply libcxx/include/module.modulemap.in:451]: Yes, if it is provided by the compiler then the compiler should provide an appropriate modulemap for that header if that's necessary.

**Comment by ldionne** [comment]
This seems to have passed the CI, so I'll merge this.

**Pull request merged**